### PR TITLE
docs theme - visual update 

### PR DESCRIPTION
**Pull request by flynnduism**
Working on a CSS update to developer.fermyon.com, ideally to complete a visual refresh before adding a new layout section.

Screen Shot 2023-07-11 at 2 00 58 AM

_light_

Screen Shot 2023-07-11 at 2 00 12 AM

_dark_

### Changes:

* uses Space Grotesk for `h1` `h2` and `h3` rather than the "Sen" typeface
* updates the Fermyon palette via SCSS variables (a bit brighter, gradients and highlights)
* adjusts the design of the documentation theme, bringing in some of the new brand styles that we have on fermyon.com
* reworks and refines some of the sidebar menu styles and topbar (moves feedback to sidebar)

### To Dos

- [x] topbar buttons and icon alignment
- [x] footer rework to avoid scroll jank
- [x] mobile adjustments and testing

### CODE REVIEW

**Comment by tpmccallum** [comment]
Hi @flynnduism 
Is it possible to also address the issue Re: [enabling search functionality on phone](https://github.com/fermyon/developer/issues/389)s here? Or is the search task out of scope for this PR?
Thanks
Tim

File: static/sass/developer-include-sidebar.scss [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+.documentation aside.menu {
+    display: flex;
+    flex-direction: column;
+    min-height: 100%;
+    border: none !important;
+    padding-bottom: 8rem !important;
+    padding-top: 0 !important;
+    background: transparent none !important;
+    overflow-x: hidden;
+    overflow-y: scroll;
```
Comment (karthik2804) [inline static/sass/developer-include-sidebar.scss:32]:
```suggestion
    overflow-y: auto;
```
Maybe adding a scroll gutter instead would be cleaner I think.

**Comment by flynnduism** [comment]
> Is it possible to also address the issue Re: https://github.com/fermyon/developer/issues/389s here? Or is the search task out of scope for this PR?

Best addressed in a follow-up (#389) as there's some changes required to do search cleanly on mobile

**Comment by flynnduism** [comment]
Preview here https://fermyon-developer-gvgttgxl.fermyon.app/

**Comment by itowlson** [comment]
The current preview has a Toggle Theme button at the bottom of the TOC, which, to borrow Bertie Wooster's turn of phrase, is "taking up space which I require for other purposes."  Toggle Theme is something that people do _once_ in the lifetime of the site and then never again - it doesn't need to take up precious TOC real estate:

image

(MS Edge, Windows 10)

**Comment by itowlson** [comment]
The previous "Login to Cloud" button is now called "LOGIN".  This seems less clear to me, especially on projects other than Cloud.

**Comment by karthik2804** [comment]
+1 to @itowlson about the new login button. 

I also find the changelog at the footer of every page a bit confusing as it all leads to the cloud changelog. Should it be called "cloud changelog"

**Comment by itowlson** [comment]
There's been a slight change to the project drop-down button.  Previously when it was dropped down the button became inactive.  Now the button takes you back to the index of the current project.  For me this is a negative because I'm used to being able to click a combo box to close it up again, effectively to cancel out of it (which is perhaps a Windows-ism).  When this did nothing, this habit was benign.  But now it means I keep accidentally dumping myself back to the project index!  (If this is intentional, it's no biggie for sure.  The Back button is right there!)

**Comment by itowlson** [comment]
I toggled dark mode and the in-page TOC is having big drama.

image

**Review by karthik2804 — APPROVED** [review]

**Comment by flynnduism** [comment]
@itowlson updates and fixes applied

**Comment by flynnduism** [comment]
Screen Shot 2023-07-20 at 1 08 36 AM

Pushed more fixes 👁️

**Comment by flynnduism** [comment]
Ready to merge this change.
There will be some follow-up edits to this that will address other visual elements seperately

* Further tweaks to color scheme - search results modal, tables (dark)
* Mobile menu improvements via #389 (search)

**Comment by itowlson** [comment]
This may be a nit, but the change theme button moves quite laggily as I resize the browser window.  E.g. if I make the browser window wider, the theme button moves with the login button (fixed offset from right hand side of window I think) _during the resize_, then after the resize it gradually moves over the period of quarter? half? a second to about 3/4 of the way along, next to the search button.

Then if I _narrow_ the window again, the theme button sometimes moves _left_ of the search button before moving gradually back to its correct position.

I don't know how to make a video of this, sorry.

**Comment by itowlson** [comment]
Also if I change theme with the page TOC open, the page TOC now changes to match the theme, but there's a slight lag, almost like a brief fade effect.

**Comment by itowlson** [comment]
Other than that (and those two things are not blockers) it all looks good - thanks for your patience dealing with all the niggles.

**Comment by karthik2804** [comment]
I think those 2 can be fixed with follow-up PRS.

**Comment by flynnduism** [comment]
Okay thanks @itowlson those are always great details to get right, and I appreciate your keen eye, it makes the difference between something being done and something being done right

I'm going to merge this so we can roll forward with these elements in the Spin Up Hub work, and will address the edits above in follow ups

**Comment by tpmccallum** [comment]
Thanks @flynnduism

**Pull request merged**